Jetchat/app/src/main/java/com/example/compose/jetchat/conversation/RecordButton.kt

```diff
@@ -0,0 +1,165 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.compose.jetchat.conversation
+
+import androidx.compose.animation.animateColor
+import androidx.compose.animation.core.Spring
+import androidx.compose.animation.core.animateFloat
+import androidx.compose.animation.core.spring
+import androidx.compose.animation.core.tween
+import androidx.compose.animation.core.updateTransition
+import androidx.compose.foundation.background
+import androidx.compose.foundation.gestures.detectDragGesturesAfterLongPress
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.aspectRatio
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.sizeIn
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Mic
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.LocalContentColor
+import androidx.compose.material3.RichTooltipBox
+import androidx.compose.material3.RichTooltipState
+import androidx.compose.material3.Text
+import androidx.compose.material3.contentColorFor
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.dp
+import kotlin.math.abs
+import kotlinx.coroutines.launch
+
+val swipeToCancelThreshold: () -> Float
+    @Composable get() = with(LocalDensity.current) { { 200.dp.toPx() } }
+
+private val verticalThreshold: () -> Float
+    @Composable get() = with(LocalDensity.current) { { 80.dp.toPx() } }
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun RecordButton(
+    recording: Boolean,
+    swipeOffset: MutableState<Float>,
+    onStartRecording: () -> Boolean,
+    onFinishRecording: () -> Unit,
+    onCancelRecording: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val transition = updateTransition(targetState = recording, label = "record")
+    val scale = transition.animateFloat(
+        transitionSpec = { spring(Spring.DampingRatioMediumBouncy, Spring.StiffnessLow) },
+        label = "record-scale",
+        targetValueByState = { rec -> if (rec) 2f else 1f }
+    )
+    val containerAlpha = transition.animateFloat(
+        transitionSpec = { tween(2000) },
```

Comment:
nit: Any reason why 2000 here and 200 in the color?

Reply:
not sure but it sure looks nice?

Reply:
good enough for me then 😄